overlord/devicestate/devicemgr.go

```diff
@@ -91,12 +91,12 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, runner *state.T
 		preseed:    snapdenv.Preseeding(),
 	}
 
-	modeEnv, err := boot.ReadModeenv("")
-	if err != nil && !os.IsNotExist(err) {
+	modeEnv, err := m.readMaybeModeenv()
```

Comment:
```suggestion
	modeEnv, err := m.maybeReadModeenv()
```

Reply:
since this was missed, I proposed this in https://github.com/snapcore/snapd/pull/8361